library/core/src/slice/cmp.rs

```diff
@@ -257,3 +257,27 @@ impl SliceContains for i8 {
         memchr::memchr(byte, bytes).is_some()
     }
 }
+
+macro_rules! impl_slice_contains {
+    ($($t:ty),*) => {
+        $(
+            impl SliceContains for $t {
+                #[inline]
+                fn slice_contains(&self, arr: &[$t]) -> bool {
+                    // Make our LANE_COUNT 4x the normal lane count. The compiler will nicely unroll it.
+                    const LANE_COUNT: usize = 4 * (128 / (mem::size_of::<$t>() * 8));
+                    // SIMD
+                    for chunk in arr.chunks_exact(LANE_COUNT){
+                        if chunk.iter().fold(false, |acc, x| acc | (*x == *self)) {
+                            return true;
+                        }
+                    }
+                    // Scalar remainder
+                    return arr.chunks_exact(LANE_COUNT).remainder().iter().any(|x| *x == *self);
```

Comment:
I think it would be nicer to store `ChunksExact` in a local variable and re-use it here to make it more obvious what's happening.